### PR TITLE
multipart filter option

### PR DESCRIPTION
**Pull request by Aks-4125**
To avoid multipart data in logcat.

based on [this](https://stackoverflow.com/questions/48661623/okhttp-logging-interceptor-disable-multipart-data) & [this](https://stackoverflow.com/questions/38563394/okhttp-loggin-interceptor-in-multipart) problem

By setting `interceptor.setShowMultipartLog(false);` we can see actual request data in log cat.

How to apply?

       `HttpLoggingInterceptor interceptor= new HttpLoggingInterceptor ();
        logging.setShowMultipartLog(false); // set true to display multipart data (by default true)
        httpClient.addInterceptor(interceptor);`

Output in logcat:

D/OkHttp: --> POST api-end-point
D/OkHttp: Content-Type: application/json; charset=UTF-8
D/OkHttp: Content-Length: 7556444
D/OkHttp: Multipart length transmitted->7556444
D/OkHttp: --> END POST (7556444-byte body)

### CODE REVIEW

File: okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java [hunk 1]
```diff
@@ -136,6 +138,23 @@ public Level getLevel() {
     return level;
   }
 
+  /**
+   *
+   * @return check mutipart logging is enabled or not
```
Comment (yschimke) [inline okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java]: nit: typo mutipart

File: okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java [hunk 2]
```diff
@@ -136,6 +138,23 @@ public Level getLevel() {
...
+   */
+  public boolean isShowMultipartLog() {
+    return showMultipartLog;
+  }
+
+  /**
+   *
+   * @param showMultipartLogging set to false to display multipart length in integer instead of non-plain text (by default true)]
```
Comment (yschimke) [inline okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java]: Can you clean up these javadoc, put most of the comment to the body and less in the param and return.

Comment (Aks-4125) [reply okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java]: sure..

**Comment by yschimke** [comment]
I'm not clear on this diff what makes it related to multipart response?  Doesn't it apply more generally?  I assume you want to avoid content that is ascii at the start but then becomes binary in a second part?

Please add some tests as well.

**Comment by Aks-4125** [comment]
That does skip multipart response of type ascii/binary.

**Comment by yschimke** [comment]
@Aks-4125 how can I help?

**Comment by yschimke** [comment]
Closing for now, please reopen if it is still active.

**Pull request closed without merging**